### PR TITLE
snap-korf: new package

### PR DESCRIPTION
**Pull request by akthoma**
Adding new package for snap-korf.

### CODE REVIEW

File: var/spack/repos/builtin/packages/snap-korf/package.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    url      = "http://korflab.ucdavis.edu/Software/snap-2013-11-29.tar.gz"
+
+    version('2013-11-29', 'dfdf48e37cdb32af4eecd9201506b6e3')
+
+    depends_on('perl', type=('build', 'run'))
+    depends_on('boost')
+    depends_on('sqlite')
+    depends_on('sparsehash')
+
+    conflicts('%gcc@5:')
```
Comment (junghans) [inline var/spack/repos/builtin/packages/snap-korf/package.py]: add `when='@2013-11-29'`.

File: var/spack/repos/builtin/packages/snap-korf/package.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        install('forge', prefix.bin)
+        install('depend', prefix.bin)
+        install('exonpairs', prefix.bin)
+        install('hmm-info', prefix.bin)
+
+        files = glob.iglob('*.pl')
+        for file in files:
+            install(file, prefix.bin)
+
+        install_tree('Zoe', prefix.Zoe)
```
Comment (junghans) [inline var/spack/repos/builtin/packages/snap-korf/package.py]: @adamjstewart : Do `prefix.Zoe` automatically get defined?

File: var/spack/repos/builtin/packages/snap-korf/package.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    conflicts('%gcc@5:')
+
+    def install(self, spec, prefix):
+        mkdirp(prefix.bin)
+        install('snap', prefix.bin)
+        install('fathom', prefix.bin)
+        install('forge', prefix.bin)
+        install('depend', prefix.bin)
+        install('exonpairs', prefix.bin)
+        install('hmm-info', prefix.bin)
```
Comment (junghans) [inline var/spack/repos/builtin/packages/snap-korf/package.py]: Can you put these in a loop?

File: var/spack/repos/builtin/packages/snap-korf/package.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        for file in files:
+            install(file, prefix.bin)
+
+        install_tree('Zoe', prefix.Zoe)
+        install_tree('HMM', prefix.HMM)
+        install_tree('DNA', prefix.DNA)
+
+    def setup_environment(self, spack_env, run_env):
+        run_env.set('ZOE', self.prefix)
+        run_env.prepend_path('PATH', self.prefix)
```
Comment (junghans) [inline var/spack/repos/builtin/packages/snap-korf/package.py]: `self.prefix.bin`?

Comment (akthoma) [reply var/spack/repos/builtin/packages/snap-korf/package.py]: No, this is how I want it. `PATH` is already going to be set to `self.prefix.bin` and I want to prepend the path to also include `self.prefix`.

File: var/spack/repos/builtin/packages/snap-korf/package.py (hunk 2 above)
Comment (akthoma) [reply var/spack/repos/builtin/packages/snap-korf/package.py]: Yeah, it works. Quoted from @adamjstewart in #4770:

>Yep, it works with any valid Python variable. So basically, . and - don't work, and neither do special characters. But letters, numbers, and underscores should all work!

Also, it is being set in the module file created by the install:
```
setenv("ZOE", "/opt/rit/app/spack/linux-rhel7-x86_64/gcc-4.8.5/snap-korf-2013-11-29-iehq347wm4cwlfph3vgrbe2ywb4beokf")
```

File: var/spack/repos/builtin/packages/snap-korf/package.py (hunk 4 above)
Comment (junghans) [reply var/spack/repos/builtin/packages/snap-korf/package.py]: Ok

File: var/spack/repos/builtin/packages/snap-korf/package.py (hunk 2 above)
Comment (junghans) [reply var/spack/repos/builtin/packages/snap-korf/package.py]: ok

File: var/spack/repos/builtin/packages/snap-korf/package.py (hunk 3 above)
Comment (akthoma) [reply var/spack/repos/builtin/packages/snap-korf/package.py]: Sure, if you don't mind *.o and *.c files that the program doesn't use going in `prefix.bin` as well. The files without an extension being installed above are located in the same directory as *.c, *.o, *.pl files, as well as `Makefile` and `00README`; if I put them in a loop I don't know a good way to get only the desired suffix-less and *.pl files into `prefix.bin`.

Comment (junghans) [reply var/spack/repos/builtin/packages/snap-korf/package.py]: Sorry, I meant something like:
```
progs = [ 'forge', 'depend', 'exonpairs', ...]
for p in progs:
   install(p, prefix.bin)
```

**Comment by akthoma** [comment]
bump

**Comment by scheibelp** [comment]
It appears all review requests were addressed so I merged this. @junghans let me know if that was premature.

**Pull request merged**